internal/controller/cluster_monitoring_controller.go

```diff
@@ -89,18 +98,24 @@ func SetupClusterMonitoringReconciler(mgr manager.Manager, conf config.Config) e
 		ManagementCluster: conf.ManagementCluster,
 	}
 
+	alloyRulesService := rules.Service{
+		Client:          managerClient,
+		AlloyAppVersion: conf.AlloyAppVersion,
+	}
+
 	r := &ClusterMonitoringReconciler{
 		Client:                     managerClient,
 		ManagementCluster:          conf.ManagementCluster,
 		HeartbeatRepository:        heartbeatRepository,
 		PrometheusAgentService:     prometheusAgentService,
+		AlloyRulesService:          alloyRulesService,
```

Comment:
Why is this resource bound to the cluster controller ? I see it using tenants as source for generating the alloy rule confg, but I don't see the relation to clusters here.

Reply:
It runs for the mc only so it's in the cluster controller but I see your point

Reply:
I think it would be better suited in the grafana organization controller

Reply:
I'll test it this evening then :)

Reply:
Done